system_tests/seqfeed_test.go

```diff
@@ -250,3 +261,101 @@ func TestLyingSequencer(t *testing.T) {
 func TestLyingSequencerLocalDAS(t *testing.T) {
 	testLyingSequencer(t, "files")
 }
+
+func testBlockHashComparison(t *testing.T, blockHash *common.Hash, mustMismatch bool) {
+	logHandler := testhelpers.InitTestLog(t, log.LvlTrace)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	backlogConfiFetcher := func() *backlog.Config {
+		return &backlog.DefaultTestConfig
+	}
+	bklg := backlog.NewBacklog(backlogConfiFetcher)
+
+	wsBroadcastServer := wsbroadcastserver.NewWSBroadcastServer(
+		newBroadcasterConfigTest,
+		bklg,
+		412346,
+		nil,
+	)
+	err := wsBroadcastServer.Initialize()
+	if err != nil {
+		t.Fatal("error initializing wsBroadcastServer:", err)
+	}
+	err = wsBroadcastServer.Start(ctx)
+	if err != nil {
+		t.Fatal("error starting wsBroadcastServer:", err)
+	}
+	defer wsBroadcastServer.StopAndWait()
+
+	port := wsBroadcastServer.ListenerAddr().(*net.TCPAddr).Port
+
+	builder := NewNodeBuilder(ctx).DefaultConfig(t, true)
+	builder.nodeConfig.Feed.Input = *newBroadcastClientConfigTest(port)
+	cleanup := builder.Build(t)
+	defer cleanup()
+	testClient := builder.L2
+
+	// related to:
+	// - builder.L2Info.PrepareTx("Owner", "User2", builder.L2Info.TransferGas, big.NewInt(1e12), nil)
+	userAccount := "User2"
+	txHash := common.HexToHash("0x633f62b463cc0e52d842406995fb590654db40aace77bfca863ba0e8d2290f97")
+	poster := common.HexToAddress("0xa4b000000000000000000073657175656e636572")
+	l2msg := []byte{4, 2, 248, 111, 131, 6, 74, 186, 128, 128, 132, 11, 235, 194, 0, 131, 122, 18, 0, 148, 12, 112, 159, 52, 15, 11, 178, 227, 97, 34, 158, 52, 91, 126, 38, 153, 157, 9, 105, 171, 133, 232, 212, 165, 16, 0, 128, 192, 1, 160, 75, 109, 200, 183, 223, 114, 85, 128, 133, 94, 26, 103, 145, 247, 47, 0, 114, 132, 133, 234, 222, 235, 102, 45, 2, 109, 83, 65, 210, 142, 242, 209, 160, 96, 90, 108, 188, 197, 195, 43, 222, 103, 155, 153, 81, 119, 74, 177, 103, 110, 134, 94, 221, 72, 236, 20, 86, 94, 226, 94, 5, 206, 196, 122, 119}
```

Comment:
let's generate these instead of hardcoding them.

create the transaction
poster is: l1pricing.BatchPosterAddress
You can turn a list of transactions + errors to an l2message usin messageFromTxes in executionEngine (it's o.k to make it public)

Reply:
Done.